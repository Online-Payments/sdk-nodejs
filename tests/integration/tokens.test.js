const sdkInstance = require('./__fixture__');
const config = require('./__fixture__/config');
const stubs = require('./__stubs__');

describe('Tokens', () => {
  describe('Promises', () => {
    test('it can get a token', async () => {
      const sdkResponse = await sdkInstance.tokens.getToken(config.merchantId, 'TOKEN_ID', {});
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });

    test('it can create a token', async () => {
      const sdkResponse = await sdkInstance.tokens.createToken(config.merchantId, stubs.createToken, {});
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });

    test('it can remove a token', async () => {
      const sdkResponse = await sdkInstance.tokens.removeToken(config.merchantId, 'TOKEN_ID', {});
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });
  });

  describe('Callback', () => {
    test('it can get a token', (done) => {
      sdkInstance.tokens.getToken(config.merchantId, 'TOKEN_ID', {}, (error, sdkResponse) => {
        expect(error).toBeNull();
        expect(sdkResponse.isSuccess).toBe(true);
        expect(sdkResponse).toMatchSnapshot();
        done();
      });
    });

    test('it can create a token', (done) => {
      sdkInstance.tokens.createToken(config.merchantId, stubs.createToken, {}, (error, sdkResponse) => {
        expect(error).toBeNull();
        expect(sdkResponse.isSuccess).toBe(true);
        expect(sdkResponse).toMatchSnapshot();
        done();
      });
    });

    test('it can remove a token', (done) => {
      sdkInstance.tokens.removeToken(config.merchantId, 'TOKEN_ID', {}, (error, sdkResponse) => {
        expect(error).toBeNull();
        expect(sdkResponse.isSuccess).toBe(true);
        expect(sdkResponse).toMatchSnapshot();
        done();
      });
    });
  });
});
