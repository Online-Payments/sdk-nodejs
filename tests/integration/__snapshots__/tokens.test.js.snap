// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tokens Callback it can create a token 1`] = `
Object {
  "body": Object {
    "card": Object {
      "cardNumber": "string",
      "cardholderName": "string",
      "expiryDate": 529,
    },
    "externalTokenLinked": Object {
      "ComputedToken": "string",
      "GTSComputedToken": "string",
      "GeneratedToken": "string",
    },
    "isNewToken": true,
    "token": "43763dd9-6b6b-4b05-914a-5953b14dcbba",
    "tokenStatus": "UNCHANGED",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Tokens Callback it can get a token 1`] = `
Object {
  "body": Object {
    "card": Object {
      "alias": "483247XXXXXX0249",
      "data": Object {
        "cardWithoutCvv": Object {
          "cardNumber": "string",
          "cardholderName": "string",
          "expiryDate": 529,
        },
      },
    },
    "eWallet": Object {
      "alias": "customer-account@email.com",
      "customer": Object {
        "billingAddress": Object {
          "additionalInfo": "floor 9",
          "city": "Zaventem",
          "countryCode": "BE",
          "houseNumber": 3,
          "state": "string",
          "street": "Da Vinci street",
          "zip": 1930,
        },
        "companyInformation": Object {
          "name": "Customer Company Name",
        },
        "personalInformation": Object {
          "name": Object {
            "firstName": "John",
            "surname": "Doe",
          },
        },
      },
    },
    "externalTokenLinked": Object {
      "ComputedToken": "string",
      "GTSComputedToken": "string",
      "GeneratedToken": "string",
    },
    "id": "43763dd9-6b6b-4b05-914a-5953b14dcbba",
    "isTemporary": true,
    "paymentProductId": 840,
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Tokens Callback it can remove a token 1`] = `
Object {
  "body": null,
  "isSuccess": true,
  "status": 204,
}
`;

exports[`Tokens Promises it can create a token 1`] = `
Object {
  "body": Object {
    "card": Object {
      "cardNumber": "string",
      "cardholderName": "string",
      "expiryDate": 529,
    },
    "externalTokenLinked": Object {
      "ComputedToken": "string",
      "GTSComputedToken": "string",
      "GeneratedToken": "string",
    },
    "isNewToken": true,
    "token": "43763dd9-6b6b-4b05-914a-5953b14dcbba",
    "tokenStatus": "UNCHANGED",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Tokens Promises it can get a token 1`] = `
Object {
  "body": Object {
    "card": Object {
      "alias": "483247XXXXXX0249",
      "data": Object {
        "cardWithoutCvv": Object {
          "cardNumber": "string",
          "cardholderName": "string",
          "expiryDate": 529,
        },
      },
    },
    "eWallet": Object {
      "alias": "customer-account@email.com",
      "customer": Object {
        "billingAddress": Object {
          "additionalInfo": "floor 9",
          "city": "Zaventem",
          "countryCode": "BE",
          "houseNumber": 3,
          "state": "string",
          "street": "Da Vinci street",
          "zip": 1930,
        },
        "companyInformation": Object {
          "name": "Customer Company Name",
        },
        "personalInformation": Object {
          "name": Object {
            "firstName": "John",
            "surname": "Doe",
          },
        },
      },
    },
    "externalTokenLinked": Object {
      "ComputedToken": "string",
      "GTSComputedToken": "string",
      "GeneratedToken": "string",
    },
    "id": "43763dd9-6b6b-4b05-914a-5953b14dcbba",
    "isTemporary": true,
    "paymentProductId": 840,
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Tokens Promises it can remove a token 1`] = `
Object {
  "body": null,
  "isSuccess": true,
  "status": 204,
}
`;
